packages/lesswrong/server/sql/SqlFragment.ts

```diff
@@ -155,6 +155,16 @@ class SqlFragment {
         continue;
       }
 
+      // Add new case for inline fragments
+      match = line.match(/^\.\.\.\s+on\s+([a-zA-Z0-9-_]+)\s*{$/);
```

Comment:
This regex is incorrect; it allows `-` as a character inside identifiers and allows identifiers to start with digits. It's also bypassing the graphql parser in a way that seems sketchy (would fail if there was a comment on the line, for example).